lib/worker.go

```diff
@@ -79,6 +79,18 @@ func NewBaseWorker(
 	}
 }
 
+func (w *BaseWorker) ID() WorkerID {
+	return w.id
+}
+
+func (w *BaseWorker) Workload() model.RescUnit {
+	wl, err := w.impl.Workload()
+	if err != nil {
+		panic(err)
```

Comment:
Better to use `log.L().Panic`.
Besides we may add some error passing mechanism in the feature, single worker error should not affect the whole runtime.